haystack_experimental/components/generators/chat/bedrock.py

```diff
@@ -0,0 +1,148 @@
+    import haystack_integrations.components.generators.amazon_bedrock.chat.utils as original_utils
+    from haystack_integrations.components.generators.amazon_bedrock.chat.utils import (
+        _format_tool_call_message,
+        _format_tool_result_message,
+        _repair_tool_result_messages,
+    )
+
+# NOTE: The following implementation ensures that:
+# - we reuse existing code where possible
+# - people can use haystack-experimental without installing amazon-bedrock-haystack.
+#
+#
+# If amazon-bedrock-haystack is installed: all works correctly.
+#
+# If amazon-bedrock-haystack is not installed:
+# - haystack-experimental package works fine (no import errors).
+# - AmazonBedrockChatGenerator fails with ImportError at init (due to bedrock_integration_import.check()).
+
+if not bedrock_integration_import.is_successful():
+    @component
+    class AmazonBedrockChatGenerator():
+        """
+        Experimental version of AmazonBedrockChatGenerator that allows multimodal chat messages.
+        """
+        def __init__(  # pylint: disable=too-many-positional-arguments
+            self,
+            model: str,
+            aws_access_key_id: Optional[Secret] = Secret.from_env_var(["AWS_ACCESS_KEY_ID"], strict=False),  # noqa: B008
+            aws_secret_access_key: Optional[Secret] = Secret.from_env_var(  # noqa: B008
+                ["AWS_SECRET_ACCESS_KEY"], strict=False
+            ),
+            aws_session_token: Optional[Secret] = Secret.from_env_var(["AWS_SESSION_TOKEN"], strict=False),  # noqa: B008
+            aws_region_name: Optional[Secret] = Secret.from_env_var(["AWS_DEFAULT_REGION"], strict=False),  # noqa: B008
+            aws_profile_name: Optional[Secret] = Secret.from_env_var(["AWS_PROFILE"], strict=False),  # noqa: B008
+            generation_kwargs: Optional[Dict[str, Any]] = None,
+            stop_words: Optional[List[str]] = None,
+            streaming_callback: Optional[StreamingCallbackT] = None,
+            boto3_config: Optional[Dict[str, Any]] = None,
+            tools: Optional[Union[List[Tool], Toolset]] = None,
+        ) -> None:
+            bedrock_integration_import.check() # this always fails
+
+        @component.output_types(replies=List[ChatMessage])
+        def run(
+            self,
+            messages: List[ChatMessage],
+            streaming_callback: Optional[StreamingCallbackT] = None,
+            generation_kwargs: Optional[Dict[str, Any]] = None,
+            tools: Optional[Union[List[Tool], Toolset]] = None,
+        ) -> Dict[str, List[ChatMessage]]:
+            """
+            Executes a synchronous inference call to the Amazon Bedrock model using the Converse API.
+            """
+
+            # NOTE: placeholder run method needed to make component happy
+            raise NotImplementedError("Unreachable code")
+else:
+    IMAGE_SUPPORTED_FORMATS = ["png", "jpeg", "gif", "webp"]
```

Comment:
Could we add a dev comment with a link where these supported formats are listed?

Reply:
added in https://github.com/deepset-ai/haystack-experimental/pull/307/commits/b1ed482feb23fc40fc2e3c70c6d7460ecf4cb9a1